src/java.base/windows/classes/java/net/DefaultDatagramSocketImplFactory.java

```diff
@@ -97,10 +97,9 @@ static DatagramSocketImpl createDatagramSocketImpl(boolean isMulticast)
                 throw new SocketException("can't instantiate DatagramSocketImpl");
             }
         } else {
-            if (!preferIPv4Stack && !isMulticast)
-                return new DualStackPlainDatagramSocketImpl(exclusiveBind);
-            else
-                return new TwoStacksPlainDatagramSocketImpl(exclusiveBind && !isMulticast, isMulticast);
+            // Always use TwoStacksPlainDatagramSocketImpl since we need
+            // to support multicasting at DatagramSocket level
+            return new TwoStacksPlainDatagramSocketImpl(exclusiveBind && !isMulticast, isMulticast);
```

Comment:
DualStackPlainDatagramSocketImpl will be orphaned, are you proposing to keeping it? At some point all of the old implementation will be removed anyway so maybe it will be removed then?

Reply:
I was keeping it in case we wanted to try & upgrade it to support multicasting. I don't think we should bother doing that, given that, as you say, we want at some point in the future to remove the whole old implementation. So yes - my plan was to keep it around just in case, until we remove the whole old implementation.